File: 1264-maximum-number-of-words-you-can-type/maximum-number-of-words-you-can-type.py
class Solution(object):
    def canBeTypedWords(self, text, brokenLetters):
        """
        :type text: str
        :type brokenLetters: str
        :rtype: int
        """

        stringArray = text.split()
        numOfWords = len(stringArray)

        for word in stringArray:
            if self.containsAnyCharacter(word, brokenLetters):
                numOfWords = numOfWords - 1

        return numOfWords

    def containsAnyCharacter(self, stringA, stringB):
        """
        :type stringA: str
        :type stringB: str
        :rtype: boolean
        """

        for letter in list(stringB):
            if letter in stringA:
                return True
        
        return False

File: 1264-maximum-number-of-words-you-can-type/maximum-number-of-words-you-can-type.js
/**
 * @param {string} text
 * @param {string} brokenLetters
 * @return {number}
 */
var canBeTypedWords = function(text, brokenLetters) {
    
    let words = text.split(" ");
    let numOfWords = words.length;

    for (let i = 0; i < words.length; i++) {
        if (containsAnyCharacter(words[i], brokenLetters)) numOfWords--; 
    }

    return numOfWords;
};

var containsAnyCharacter = function(stringA, stringB) {
  for (let i = 0; i < stringB.length; i++) {
    if (stringA.includes(stringB[i])) return true;
  }

  return false;
}